SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -1408,9 +1408,18 @@ void USpatialReceiver::ApplyComponentDataOnActorCreation(Worker_EntityId EntityI
 		Channel.CreateSubObjects.Add(TargetObject.Get());
 	}
 
+	FString TargetObjectPath = TargetObject->GetPathName();
 	ApplyComponentData(Channel, *TargetObject, Data);
 
-	OutObjectsToResolve.Add(ObjectPtrRefPair(TargetObject.Get(), TargetObjectRef));
+	if (TargetObject.IsValid())
+	{
+		OutObjectsToResolve.Add(ObjectPtrRefPair(TargetObject.Get(), TargetObjectRef));
+	}
+	else
+	{
+		UE_LOG(LogSpatialReceiver, Warning, TEXT("Actor subobject got invalidated after applying component data! Subobject: %s"),
```

Comment:
Feel like we could maybe make this more readable for Unreal developers, but not sure how.
"Are you deleting a subobject in an OnRep"? And theoretically we don't know if it's really an invalid flow.

Reply:
The reasoning to make this a warning is that we don't know if this might cause issues with something else. I'll create a follow up ticket to do a proper investigation.